### PR TITLE
Fixed visibility of atomic properties as required by the new JVM compiler plugin

### PR DESCRIPTION
**Pull request by mvicsokolova**
The new atomicfu JVM compiler plugin (since 1.9.20) will explicitly require atomic properties to be private or internal (or be members of private or internal classes). These fixes are required now in the aggregate build to merge changes in the new jvm plugin, so this commit is added to develop first.

### CODE REVIEW

**Review by qwwdfsad — COMMENTED** [review]
Please don't forget to put https://github.com/Kotlin/kotlinx.coroutines/labels/kotlin-merge-blocker on the PR and mention the corresponding issue (whether it's KT/atomicfu GH or just a PR), so we can keep track of the change later

**Comment by qwwdfsad** [comment]
>Should we put kotlin-merge-blocker label on PRs that are merged into develop?

Yes, on everything that breaks aggregate. This is also a signal for release manager to proceed with rebase, see https://jetbrains.team/p/kti/repositories/kotlin-teamcity-build/files/.teamcity/common/buildtypes/tests/userprojects/mainAggregate/README.md
Also, a reviewer from release team us required

**Review by qwwdfsad — APPROVED** [review]
LGTM, but please merge only after every step from the aggregate process is fulfilled

**Comment by mvicsokolova** [comment]
Ok, thanks! Though this should not break the aggregate build.
All the changes in the JVM compiler plugin are still in MR (https://jetbrains.team/p/kt/reviews/10579/timeline) and not merged because the aggregate is red and needs these changes in the `kotlin-community/*` branch.

**Review by mshishkina — APPROVED** [review]

**Pull request merged**
